README.md

```diff
@@ -4,204 +4,430 @@ Daniel Ehrenberg, Igalia. Stage 3
-- Unary `+` followed by an expression is always either a Number, or results in throwing. For this reason, unfortunately, `+` on a BigInt needs to throw, rather than being symmetrical with `+` on Number: Otherwise, previously "type-declared" asm.js code would now be polymorphic.
-- `|0` always returns a Number in int32 range, or throws. This proposal maintains that, as it would throw on a BigInt for being a mixed operand type.
-- `Math.fround` always returns a Number in float32 range, or throws. This proposal would throw if `Math.fround` is called with a BigInt, preserving the property.
+Attempting to convert a fractional value to a `BigInt` throws an exception both when the value is represented as an `Number` and a `String`.
+
+```js
+BigInt(1.5)
+// ↪ RangeError: The number 1.5 is not a safe integer and thus cannot be converted to a BigInt
+
+BigInt('1.5')
+// ↪ SyntaxError: Cannot convert 1.5 to a BigInt
+
+```
+
+`Math.round`, `Math.fround` and `|` throw exceptions when called with a `BigInt` since these functions are defined as returning a `Number`.
+
+```js
+
+Math.round(1n)
+// ↪ TypeError: Cannot convert a BigInt value to a number
+
+Math.fround(1n)
+// ↪ TypeError: Cannot convert a BigInt value to a number
+
+1n|0
+// ↪ TypeError: Cannot mix BigInt and other types, use explicit conversions
+
+```
+
+### Usage Recommendation
+
+It is recommended to continue using `Number` for code which will only encounter values under 2<sup>53</sup>.
+
+Reserve `BigInt` for cases where large values are expected.
+
+## About the Proposal
+
+### Motivation: Why Do We Need Such Big Numbers?
+
+There are a number of cases in JavaScript coding where integers larger than 2<sup>53</sup> come up — both instances where signed or unsigned 64-bit integers are needed and times where we may want integers even larger than 64-bits.
+
+#### 64-bit Use Cases
+
+Often, other systems with which Javascript interacts provides data as 64-bit integers, which lose precision when coerced to Javascript numbers.
+
+These might come when reading certain machine registers or wire protocols or using protobufs or JSON documents that have GUIDs generated by 64-bit systems in them — including things like credit card or account numbers.
+
+In node, `fs.stat` may give some data as 64-bit integers, which [has caused issues already](https://github.com/nodejs/node/issues/12115):
+
+```js
+fs.lstatSync('one.gif').ino
+// ↪ 9851624185071828
+
+fs.lstatSync('two.gif').ino
+// ↪ 9851624185071828, duplicate, but different file!
+```
 
-Analogously, `>>> 0` always returns a Number in uint32 range, throwing as `>>>` is not supported on BigInt at all. Note: asm.js itself does not require this property, as `>>>` may be an overloaded operator, and `|0` is used for all `int` parameter declarations, but `>>> 0` is a common idiom to achieve this property in JavaScript code.
+Finally, 64-bit integers enable higher resolution — _nanosecond!_ — timestamps.
```

Comment:
BigInt is used in the new proposed Date library, the [temporal proposal](https://github.com/tc39/proposal-temporal).